src/core/fonts.js

```diff
@@ -461,6 +461,20 @@ function adjustMapping(charCodeToGlyphId, hasGlyph, newGlyphZeroId) {
       glyphId = newGlyphZeroId;
     }
 
+    // Fix for bug 1778484:
+    // The charcodes are moved into the private area in to fix some rendering
+    // issues (https://github.com/mozilla/pdf.js/pull/9340) but when printing
+    // into pdf the generated will contains wrong chars. We can avoid that in
+    // adding the unicode to the cmap and the print backend will then map the
+    // glyph ids to the correct unicode.
+    let unicode = toUnicode.get(originalCharCode);
+    if (typeof unicode === "string") {
+      unicode = unicode.codePointAt(0);
+    }
+    if (unicode) {
+      newMap[unicode] = glyphId;
```

Comment:
This will affect glyph-mapping in a way that is likely not intended, and could very well cause issues; does this actually pass all tests?
These changes thus go a lot further than only the `cmap`, according to the commit message, and note how we're using `newMapping.charCodeToGlyphId` as part of the following code that builds a `seac`-map:
https://github.com/mozilla/pdf.js/blob/0205597bd09fca95684a8357b275f543d940f66a/src/core/fonts.js#L3009-L3078

What I'm wondering is thus if you don't want to leave the existing `newMap` alone here and rather introduce a new "map" (e.g. named `completeMap` or something better) that contains the "full"-data and return that from this function?
Then you can pass that one in when building e.g. the `cmap`-table, without taking the risk of affecting any other pre-existing code.

Reply:
I didn't run the tests: I wanted to have your opinion before.

Reply:
A question I was asking to myself in writing the patch: could we use this font in text layer ? or could you imagine some painful drawbacks ?

Reply:
> could we use this font in text layer ?

I'd advise against that, since it'll cause (lots of) problems in the general case.

> or could you imagine some painful drawbacks ?

There's no guarantee that a font ever contains *all* of the glyphs that correspond to the data in /ToUnicode. One trivial example is *ligatures*, since a font doesn't necessarily contain the individual glyphs that make up one.
Another example is OCRed documents, where the glyphs may be completely empty and thus have no usable size.